README.md

```diff
@@ -603,11 +603,22 @@ To release a new version, follow these steps:
 
 4. Attach the [binary artifacts built in CI](https://circleci.com/docs/artifacts/#artifacts-overview) to the release. Use `scripts/fetch_ci_binaries.py` to fetch all artifacts of a CI workflow.
 
+### Development - External PRs
+
+Read more about how to review PRs in [the guidelines](.github/CONTRIBUTING.md#review).
+
+Our CI/CD platform (CircleCI) does not have the option to trigger the workflow on click. So once a PR is reviewed and looks like its workflow could pass, you can either accept & merge it blindly (which shall trigger the workflow on the target branch), or use the following procedure to trigger it:
+
+- Read [this](https://stackoverflow.com/questions/5884784/how-to-pull-remote-branch-from-somebody-elses-repo).
+- Check out the contributor's branch.
+- Create your own branch from it.
```

Comment:
there is an easier way instead of creating a branch from the contributor's one. When the contributor's branch is checked out with the `--track` option. Then update the push url for the specific remote.
For example:
```bash
git remote add coworker git://path/to/coworkers/repo.git
git fetch coworker
git checkout --track coworker/foo

git remote set-url --push coworker git@github.com:0xSpaceShard/starknet-devnet-rs.git
```

Reply:
Indeed this happens when there is the same branch name. So a more robust example is:
```bash
git remote add coworker git://path/to/coworkers/repo.git
git fetch coworker
git checkout -b coworker/foo coworker/foo

git remote set-url --push coworker git@github.com:0xSpaceShard/starknet-devnet-rs.git
git push coworker HEAD
```

Reply:
or another name could be used for the local branch name as long as it is not already on remote

Reply:
What if the coworker adds more commits to `coworker/foo`? Will `git pull` work?

I have to say this is very similar to the approach I originally suggested (yours shows the substitutable commands though).

Reply:
While reviewing another external PR (#444), I ran this:
```
$ git remote add lordshashank git@github.com:lordshashank/starknet-devnet-rs.git
$ git fetch lordshashank
$ git checkout -b lordshashank/lite-mode-impl lordshashank/lite-mode-impl
$ git remote set-url --push lordshashank git@github.com:0xSpaceShard/starknet-devnet-rs.git
```
Then I ran:
```
$ git checkout main
$ git checkout lordshashank/lite-mode-impl
warning: refname 'lordshashank/lite-mode-impl' is ambiguous.
```

There is an ambiguity warning, which is no wonder since on `git log` the branch name is shown twice
![image](https://github.com/0xSpaceShard/starknet-devnet-rs/assets/28579396/8da14570-bf7d-4b82-984a-ecc5da479dfd)

Reply:
> What if the coworker adds more commits to `coworker/foo`? Will `git pull` work?
> 
> 
> 
> I have to say this is getting very similar to the approach I originally suggested (yours shows the substitutable commands though).

Yes git pull works.
The approach that I suggested is based on the link you send. The steps with creating a branch, then merging the remote branch to your branch, then pushing your branch to remote to run the CI are avoided by changing the remote url of "coworker" branch

Reply:
> are avoided by changing the remote url of "coworker" branch

That's what confused me - I thought that changing the remote url would affect pulling, too. So does that mean that after these steps `git pull` uses one origin and `git push` uses another?

Reply:
Yes pulling and pushing to different urls

Reply:
Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/454/commits/cfcd0086c85d480e73f03f53063441110c7d137e